packages/engine/src/map/MapUpdateSystem.ts

```diff
@@ -82,21 +86,21 @@ export default async function MapUpdateSystem(world: World): Promise<System> {
 
     if (mapState.needsUpdate) {
       // Perf hack: Start with an empty array so that any children that have been purged or that do not meet the criteria for adding are implicitly removed.
-      const subSceneChildren = []
+      object3dComponent.value.children.length = 0
```

Comment:
just for my understanding: was this change for performance or did you need to maintain other references to this array (or both)?

Reply:
That was really just for simplicity, I assumed it would help with performance but didn't test that specifically.